sentry-anyhow/src/lib.rs

```diff
@@ -57,36 +57,13 @@ pub fn capture_anyhow(e: &anyhow::Error) -> Uuid {
 
 /// Hub extension methods for working with `anyhow`.
 pub trait AnyhowHubExt {
-    /// Captures an `anyhow::Error` on a specific hub.
+    /// Captures an [`anyhow::Error`] on a specific hub.
     fn capture_anyhow(&self, e: &anyhow::Error) -> Uuid;
 }
 
 impl AnyhowHubExt for Hub {
     fn capture_anyhow(&self, e: &anyhow::Error) -> Uuid {
-        self.capture_error(&AnyhowError(e))
-    }
-}
-
-// `anyhow::Error` itself does not impl `std::error::Error`, because it would
-// be incoherent. This can be worked around by wrapping it in a newtype
-// which impls `std::error::Error`.
-// Code adopted from: https://github.com/dtolnay/anyhow/issues/63#issuecomment-590983511
-struct AnyhowError<'a>(&'a anyhow::Error);
-
-impl fmt::Debug for AnyhowError<'_> {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        self.0.fmt(fmt)
-    }
-}
-
-impl fmt::Display for AnyhowError<'_> {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        self.0.fmt(fmt)
-    }
-}
-
-impl Error for AnyhowError<'_> {
-    fn source(&self) -> Option<&(dyn Error + 'static)> {
-        self.0.source()
+        let e: &dyn std::error::Error = e.as_ref();
+        self.capture_error(e)
```

Comment:
Just curious, why do you need to annotate types here?

Reply:
can’t remember exactly, but I needed it to make rust happy.